python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -2934,12 +2934,11 @@ def export_records(self, refs: Iterable[DatasetIdRef]) -> Mapping[str, Datastore
         # Docstring inherited from the base class.
         exported_refs = list(self._bridge.check(refs))
         ids = {ref.id for ref in exported_refs}
-        records: defaultdict[DatasetId, defaultdict[str, List[StoredDatastoreItemInfo]]] = defaultdict(
-            lambda: defaultdict(list), {id: defaultdict(list) for id in ids}
-        )
+        records: dict[DatasetId, dict[str, list[StoredDatastoreItemInfo]]] = {id: {} for id in ids}
```

Comment:
Isn't it all a bit annoying that we can't use defaultdict specifically for this class?

Reply:
I think we can use `defaultdict` but we cannot use lambdas or nested methods. Defining a module-level factory method just for this feels more complicated than just sticking to good old `setdefault`.